source/settings.rst

```diff
@@ -215,25 +215,62 @@ depending on whether or not they are distributed with that repository.
+          If no user defined image could be found, a fallback images is used.
+        - ``None`` - If selected, no user-defined images are loaded and the fallback is evaluated immediately.
+        - ``Custom`` - An advanced mode that allows you to set one or more custom avatar provider services (e.g. Libravatar) by providing URL
+          templates.
 
-      .. setting:: No image service
+          URL Template Syntax
+            The URL template syntax consists of regular URLs to avatar images, that can be enriched with variables, which are substituted before
+            evaluation. Those variables are encoded using curly brackets ``{}`` and can be used like this: ``https://example.avatar.service/u/{email}/avatar.png``.
+            If a request fails (http 400 and 500 errors) or does not provide a valid image, the next URL is used. More URLs can be specified by chaining them
+            together with semicolons (";") like so: ``http://provider1.com/{sha1}.png;http://provider2.com/{sha1}.png``. If all custom URLs fail to provide
+            an avatar image, the gitextensions internal fallback mechanism will provide one for that user.
 
-        If the author has not set up their own image, then gravatar can return an image based on one of these services.
+            The variable names are case insensitive. If a variable is not found (for example because of typo or it does not exist), it is substituted
+            with an empty string, so the resulting URL never contains the curly brackets.
+
+            The following variables are currently supported:
+
+            - ``name`` - The name of the commit author (git config ``user.name``). Special characters are URL encoded.
+            - ``email`` - The email address of the commit author (git config ``user.email``). Special characters are URL encoded.
+            - ``md5`` - A lowercase hex representation of the MD5 hash of the normalized (all characters lowercase)
+              email address (without URL encoding). This hash is compatible with Gravatar and thus compatible with a lot of similar services.
+            - ``sh1`` - Like the ``md5`` variable but with SHA1 as hash algorithm.
+            - ``sha256`` - Like the ``md5`` variable but with SHA256 as hash algorithm.
+            - ``imagesize`` - Represents the requested avatar size in pixels.
+
+            A complete working configuration might look something like this:
+            ``https://www.libravatar.org/avatar/{md5}?s={imageSize}&default=404;https://avatar.tobi.sh/{md5}?size={imageSize}``
+
+      .. setting:: Fallback generated avatar style
+        :id: avatar-fallback
+
+        The configured fallback determines how authors without a user-defined avatar are rendered. Besides ``Author Initials`` all other options are
+        provided by Gravatar. Details about their fallback modes can be found here https://en.gravatar.com/site/implement/images/ in the section "Default Image".
```

Comment:
Is only Gravatar using other than Author Initials? How to rewrite this?

Reply:
Sorry I don't quite get what you mean. Of the currently available / implemented fallback options
![image](https://user-images.githubusercontent.com/2495113/106068213-825d9f00-6100-11eb-958b-29fa053f302e.png)
only `Author Initials` is done locally (by gitextensions)

For the other options we query gravatar. Or strictly speaking we make sure that gravatar is queried with a set fallback (so we don't have to send a query to gravatar twice)